cm2350/ppc_peripherals.py

```diff
@@ -78,6 +78,20 @@ def __init__(self, emu, devname):
                 raise Exception('ERROR: duplicate project config entries for peripheral %s:\n%s' %
                         (devname, emu.vw.config.project.reprConfigPaths()))
 
+    def __del__(self):
+        self.shutdown()
+
+    def shutdown():
+        """
+        Standard "module" peripheral shutdown/cleanup function. This is called
+        when deleting/shutting down the emulator to clean up any allocated or
+        opened resources. During a normal shutdown this isn't critical but it
+        is helpful to have a clean shutdown for unit testing.
+
+        This function is required to be implemented by all peripherals.
+        """
+        raise NotImplementedError('%s class should implement this method' % self.__class__.__name__)
+
```

Comment:
Ahhh, so we're on the same page.  If you're worried about  other things being modules, you can create a base-class Module or somesuch, and put these things there.  Then have Peripheral subclass that.

Point of interest, I believe this should have a `(self)` not `()`

Reply:
doh! you are correct that should have `(self)`